docs/docs/examples/workflow/citation_query_engine.ipynb

```diff
@@ -0,0 +1,507 @@
+{
```

Comment:
<pre class="ql-syntax" spellcheck="false">Line #30.        async def ingest(
</pre><p>i have a hot take that ingestion really should live outside the workflows. it's completely decoupled and encourages poor developer practices to stuff two disparate things into the same workflow, especially since we don't have an explicit UX for specifying an entry point </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/run-llama/llama_index/pull/15372/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='11'/>

Reply:
I see all the workflow examples, we made ingestion part of workflows? Should I decouple it?

Reply:
yeah i know...could we put it outside anyhow? that's my vote

Reply:
Yeah I have updated the notebook.